samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/ItemBean.java

```diff
@@ -1109,137 +1064,74 @@ else if (type.equals(TypeFacade.MULTIPLE_CORRECT_SINGLE_SELECTION.toString())) {
-     //   String newvalue = (String) event.getNewValue();
-          String newvalue = this.getAdditionalChoices();
-        ArrayList list = getMultipleChoiceAnswers(); // get existing list
+        String newvalue = getAdditionalChoices();
+        List<AnswerBean> list = getMultipleChoiceAnswers();
         if (list!=null) {
-                // add additional answer bean
-           int currentsize = list.size();
-           int newlength = currentsize+ new Integer(newvalue).intValue();
-           if (newlength<=26){
-              for (int i=currentsize; i<newlength; i++){
-                  AnswerBean answerbean = new AnswerBean();
-                  answerbean.setSequence( Long.valueOf(i+1));
-                  answerbean.setLabel(AnswerBean.getChoiceLabels()[i]);
-                  list.add(answerbean);
-
+            // add additional answer bean
+            int currentsize = list.size();
+            int newlength = currentsize+ new Integer(newvalue).intValue();
+            if (newlength<=26){
+                for (int i=currentsize; i<newlength; i++){
+                    AnswerBean answerbean = new AnswerBean();
+                    answerbean.setSequence( Long.valueOf(i+1));
+                    answerbean.setLabel(AnswerBean.getChoiceLabels()[i]);
+                    list.add(answerbean);
                 }
-              setMultipleChoiceAnswers(list);
-              setAdditionalChoices("0");
-
-              // if mcmc, need to set corrAnswers 
-              if (TypeFacade.MULTIPLE_CORRECT.toString().equals(this.itemType) || TypeFacade.MULTIPLE_CORRECT_SINGLE_SELECTION.toString().equals(this.itemType)) {
-                 ArrayList corranswersList = ContextUtil.paramArrayValueLike("mccheckboxes");
-                 int corrsize = corranswersList.size();
-                 int counter = 0;
-                 String[] corrchoices = new String[corrsize];
-                 Iterator iter = corranswersList.iterator();
-                 while (iter.hasNext()) {
-
-                   String currentcorrect = (String) iter.next();
-	           corrchoices[counter]= currentcorrect; 
-                   counter++;
-                 }
-                 this.setCorrAnswers(corrchoices);
-             }
-           }
-           else
-	       {
-		   //print error
-               FacesContext context=FacesContext.getCurrentInstance();
-               context.addMessage(null,new FacesMessage(RB_AUTHOR_MESSAGES.getString("MCanswer_outofbound_error")));
-	       }
-    
-       }
+                setMultipleChoiceAnswers(list);
+                setAdditionalChoices("0");
+
+                // if mcmc, need to set corrAnswers 
+                if (TypeFacade.MULTIPLE_CORRECT.toString().equals(this.itemType) || TypeFacade.MULTIPLE_CORRECT_SINGLE_SELECTION.toString().equals(this.itemType)) {
+                    ArrayList corranswersList = ContextUtil.paramArrayValueLike("mccheckboxes");
```

Comment:
Use `List<T>` instead of concrete class `ArrayList`

Reply:
I just added tabulation on that bit and didn't realize. Thanks, sorting it now.